### PR TITLE
Fixes for unicode path handling

### PR DESCRIPTION
**Pull request by jnagi-intel**
### Description
Replace `path::string()` / bare `std::filesystem::path(string)` with
`PathToUTF8String` / `ToPathString` in two places that handle user-supplied
paths.

### Motivation and Context
On Windows, `path::string()` and `std::filesystem::path(std::string)` use the
system ANSI code page (CP_ACP). When a model or EPContext output file sits in a
directory with non-ASCII Unicode characters, this corrupts the path and causes:

- `ModelMetadefIdGenerator::GenerateId` — throws during session initialization
  ("No mapping for the Unicode character exists in the target multi-byte code page")
- `ModelGenOptions` (EPContext options) — constructs a garbled path, failing
  EPContext file creation with `ENOENT`

### CODE REVIEW

**Review by adrianlizarraga — COMMENTED** [review]
Would be great to add a unit test that uses non-ascii unicode characters for model compilation. Could be placed in this file: https://github.com/microsoft/onnxruntime/blob/a49da870aaaeded4b7d8bd9e937fedb2cf321493/onnxruntime/test/autoep/test_execution.cc#L778

**Comment by jnagi-intel** [comment]
> Would be great to add a unit test that uses non-ascii unicode characters for model compilation. Could be placed in this file:
> 
> https://github.com/microsoft/onnxruntime/blob/a49da870aaaeded4b7d8bd9e937fedb2cf321493/onnxruntime/test/autoep/test_execution.cc#L778

@adrianlizarraga Added tests for both the fixes.

File: onnxruntime/test/autoep/test_execution.cc [hunk 1]
```diff
@@ -828,6 +832,69 @@ TEST(OrtEpLibrary, PluginEp_GenEpContextModel_ErrorOutputModelExists_AutoGenOutp
...
+// Test that EPContext generation and inference work correctly when non-ASCII Unicode characters
+// appear in both the model file path and the EPContext output path.
+TEST(OrtEpLibrary, PluginEp_GenEpContextModel_UnicodePath) {
+  namespace fs = std::filesystem;
+
+  // Set up a Unicode working directory and model path (U+4E2D U+6587, "中文").
+  const fs::path unicode_dir{fs::path(u8"\u4e2d\u6587")};
+  fs::remove_all(unicode_dir);
+  fs::create_directories(unicode_dir);
+
```
Comment (adrianlizarraga) [inline onnxruntime/test/autoep/test_execution.cc:848]: I recommend using RAII to clean up this temporary directory, so that it is cleaned up even if the test fails:

```C++
   auto cleanup = gsl::finally([&unicode_dir] {
     std::error_code ec;
     fs::remove_all(unicode_dir, ec);
   });
```

File: onnxruntime/test/autoep/test_execution.cc [hunk 2]
```diff
@@ -828,6 +832,69 @@ TEST(OrtEpLibrary, PluginEp_GenEpContextModel_ErrorOutputModelExists_AutoGenOutp
...
+}
+
+// Test that inference works correctly when non-ASCII Unicode characters appear in the model file path.
+TEST(OrtEpLibrary, PluginEp_Inference_UnicodePath) {
+  namespace fs = std::filesystem;
+
+  const fs::path unicode_dir{fs::path(u8"\u4e2d\u6587")};
+  fs::remove_all(unicode_dir);
+  fs::create_directories(unicode_dir);
+
```
Comment (adrianlizarraga) [inline onnxruntime/test/autoep/test_execution.cc:888]: I recommend using RAII to clean up this temporary directory, so that it is cleaned up even if the test fails:

```C++
   auto cleanup = gsl::finally([&unicode_dir] {
     std::error_code ec;
     fs::remove_all(unicode_dir, ec);
   });
```

**Review by adrianlizarraga — COMMENTED** [review]
One more small issue.

File: onnxruntime/test/autoep/test_execution.cc (hunk 1 above)
Comment (jnagi-intel) [reply onnxruntime/test/autoep/test_execution.cc:848]: added

File: onnxruntime/test/autoep/test_execution.cc (hunk 2 above)
Comment (jnagi-intel) [reply onnxruntime/test/autoep/test_execution.cc:888]: added

**Review by adrianlizarraga — APPROVED** [review]
